### PR TITLE
Documentation of TMA: part 1

### PR DESCRIPTION
**Pull request by zasdfgbnm**
This is the first part of the documentation of TMA. It covers the topic:
- How does TMA looks like?
- What is a box, and what is a tile?
- What is TMA domain?
- Why we can not merge discontiguous IterDomains to form TMA domain?

The documentation is far from complete, but I'd like to split the work so that we can make progress incrementally.

### CODE REVIEW

File: doc/reading/divisibility-of-split.md
Comment (zasdfgbnm) [inline doc/reading/divisibility-of-split.md:1]: Renamed from doc/reading/merge-split-vs-split-merge.md

File: doc/dev/tma/dense-and-strided-tile.svg
Comment (naoyam) [inline doc/dev/tma/dense-and-strided-tile.svg:1]: Could you mention what each of the colors indicates?

File: doc/dev/tma.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+## Schedule
+
+In order to use TMA, we need to tell the hardware what is the dimensionality of our tensor.
+Most naively, we can make this dimensionality the size of the allocation domain.
+This naive mental model does provide us an easy way to reason about and start using TMA,
+but unfortunately it is not flexible enough and is not consistent with how we think about scheduling.
+For example, if we want to schedule a fusion containing only pure pointwise (i.e. no broadcasting) operations and all input and output tensors are contiguous,
+regardless of the actual dimensionality of the input and output tensors, in our mental model, we always
+consider this problem as a 1D problem by viewing all tensors as flattened 1D tensor.
+For this case, ideally, we should be using 1D TMA, instead of using the actual dimensionality of the tensor.
```
Comment (naoyam) [inline doc/dev/tma.md:35]: Side question: Is there any perf difference by the dimensionality?

File: doc/dev/tma/matmul-tma-domain.svg
Comment (naoyam) [inline doc/dev/tma/matmul-tma-domain.svg:1]: nit: may want to add some indication that both the TMA domains are part of A' and B'. The first time I saw this I thought each of these rectangles corresponds to a tensor, which is correct for the upper four rectangles.

**Review by naoyam — APPROVED** [review]
Thanks a lot for this writeup. So far my understanding is pretty clear.

File: doc/dev/tma.md (hunk 1 above)
Comment (zasdfgbnm) [reply doc/dev/tma.md:35]: I don't know. Never benchmarked.

File: doc/dev/tma/dense-and-strided-tile.svg
Comment (zasdfgbnm) [reply doc/dev/tma/dense-and-strided-tile.svg:1]: added

File: doc/dev/tma/matmul-tma-domain.svg
Comment (zasdfgbnm) [reply doc/dev/tma/matmul-tma-domain.svg:1]: added

**Pull request merged**